anta/tests/avt.py

```diff
@@ -0,0 +1,223 @@
+from anta.tools import get_value
+
+
+class VerifyAVTPathHealth(AntaTest):
+    """
+    Verifies the status of all Adaptive Virtual Topology (AVT) paths for all VRFs.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all AVT paths for all VRFs are active and valid.
+    * Failure: The test will fail if the AVT path is not configured or if any AVT path under any VRF is either inactive or invalid.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.avt:
+      - VerifyAVTPathHealth:
+    ```
+    """
+
+    name = "VerifyAVTPathHealth"
+    description = "Verifies the status of all AVT paths for all VRFs."
+    categories: ClassVar[list[str]] = ["avt"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show adaptive-virtual-topology path")]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyAVTPathHealth."""
+        # Initialize the test result as success
+        self.result.is_success()
+
+        # Get the command output
+        command_output = self.instance_commands[0].json_output.get("vrfs", {})
+
+        # Check if AVT is configured
+        if not command_output:
+            self.result.is_failure("Adaptive virtual topology paths are not configured.")
+            return
+
+        # Iterate over each VRF
+        for vrf, vrf_data in command_output.items():
+            # Iterate over each AVT path
+            for profile, avt_path in vrf_data.get("avts", {}).items():
+                for path, flags in avt_path.get("avtPaths", {}).items():
+                    # Get the status of the AVT path
+                    valid = flags["flags"]["valid"]
+                    active = flags["flags"]["active"]
+
+                    # Check the status of the AVT path
+                    if not valid and not active:
+                        self.result.is_failure(f"AVT path {path} for profile {profile} in VRF {vrf} is invalid and not active.")
+                    elif not valid:
+                        self.result.is_failure(f"AVT path {path} for profile {profile} in VRF {vrf} is invalid.")
+                    elif not active:
+                        self.result.is_failure(f"AVT path {path} for profile {profile} in VRF {vrf} is not active.")
+
+
+class VerifyAVTSpecificPath(AntaTest):
+    """
+    Verifies the status and type of an Adaptive Virtual Topology (AVT) path for a specified VRF.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the AVT path is active, valid, and matches the specified type (direct/multihop) for the given VRF.
+    * Failure: The test will fail if the AVT path is not configured or if the AVT path is not active, valid, or does not match the specified type for the given VRF.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.avt:
+      - VerifyAVTSpecificPath:
+          avt_paths:
+            - avt_name: CONTROL-PLANE-PROFILE
```

Comment:
need to add `vrf` here as we could have the same AVT used in multiple VRF

Reply:
ok sorry - it is missing in the example only :)

Reply:
updated the example